### ISSUE TITLE
[Feat]:

### ISSUE THREAD

**Issue by tianyi725** [appium/appium#18995]
`PS C:\Users\ty> appium driver install uiautomator2
× Installing 'uiautomator2' using NPM install spec 'appium-uiautomator2-driver'
Error: × Encountered an error when installing package: npm command 'install --save-dev --omit=peer --save-exact --global-style --no-package-lock appium-uiautomator2-driver --json' failed with code 1.

STDOUT:
{
  "error": {
    "code": 1,
    "summary": "command failed",
    "detail": "C:\\WINDOWS\\system32\\cmd.exe /d /s /c node install-npm.js\n[10:48:57] Error installing Chromedriver: read ECONNRESET\n[10:48:57] Error: read ECONNRESET\n    at Function.AxiosError.from (C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\axios\\lib\\core\\AxiosError.js:89:14)\n    at RedirectableRequest.handleRequestError (C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\axios\\lib\\adapters\\http.js:591:25)\n    at RedirectableRequest.emit (node:events:526:35)\n    at ClientRequest.eventHandlers.<computed> (C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\follow-redirects\\index.js:14:24)\n    at ClientRequest.emit (node:events:514:28)\n    at TLSSocket.socketErrorListener (node:_http_client:495:9)\n    at TLSSocket.emit (node:events:514:28)\n    at emitErrorNT (node:internal/streams/destroy:151:8)\n    at emitErrorCloseNT (node:internal/streams/destroy:116:3)\n    at processTicksAndRejections (node:internal/process/task_queues:82:21)\n[10:48:57] Downloading Chromedriver can be skipped by setting the'APPIUM_SKIP_CHROMEDRIVER_INSTALL' environment variable."
  }
}

STDERR:
npm WARN config global-style This option has been deprecated in favor of `--install-strategy=shallow`
npm WARN cleanup Failed to remove some directories [
npm WARN cleanup   [
npm WARN cleanup     'C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\@appium',
npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\ty\.appium\node_modules\appium-uiautomator2-driver\node_modules\@appium\support\node_modules\sharp\vendor\8.14.2\win32-x64'] {
npm WARN cleanup       errno: -4048,
npm WARN cleanup       code: 'EPERM',
npm WARN cleanup       syscall: 'rmdir',
npm WARN cleanup       path: 'C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\@appium\\support\\node_modules\\sharp\\vendor\\8.14.2\\win32-x64'
npm WARN cleanup     }
npm WARN cleanup   ],
npm WARN cleanup   [
npm WARN cleanup     'C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules',
npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\ty\.appium\node_modules\appium-uiautomator2-driver\node_modules\@appium\support\node_modules\sharp\vendor\8.14.2'] {
npm WARN cleanup       errno: -4048,
npm WARN cleanup       code: 'EPERM',
npm WARN cleanup       syscall: 'rmdir',
npm WARN cleanup       path: 'C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\@appium\\support\\node_modules\\sharp\\vendor\\8.14.2'
npm WARN cleanup     }
npm WARN cleanup   ],
npm WARN cleanup   [
npm WARN cleanup     'C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\@appium\\support',
npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\ty\.appium\node_modules\appium-uiautomator2-driver\node_modules\@appium\support\node_modules\sharp\vendor'] {
npm WARN cleanup       errno: -4048,
npm WARN cleanup       code: 'EPERM',
npm WARN cleanup       syscall: 'rmdir',
npm WARN cleanup       path: 'C:\\Users\\ty\\.appium\\node_modules\\appium-uiautomator2-driver\\node_modules\\@appium\\support\\node_modules\\sharp\\vendor'
npm WARN cleanup     }
npm WARN cleanup   ]
npm WARN cleanup ]
npm ERR! code 1
npm ERR! path C:\Users\ty\.appium\node_modules\appium-uiautomator2-driver\node_modules\appium-chromedriver
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node install-npm.js
npm ERR! [10:48:57] Error installing Chromedriver: read ECONNRESET
npm ERR! [10:48:57] Error: read ECONNRESET
npm ERR!     at Function.AxiosError.from (C:\Users\ty\.appium\node_modules\appium-uiautomator2-driver\node_modules\axios\lib\core\AxiosError.js:89:14)
npm ERR!     at RedirectableRequest.handleRequestError (C:\Users\ty\.appium\node_modules\appium-uiautomator2-driver\node_modules\axios\lib\adapters\http.js:591:25)
npm ERR!     at RedirectableRequest.emit (node:events:526:35)
npm ERR!     at ClientRequest.eventHandlers.<computed> (C:\Users\ty\.appium\node_modules\appium-uiautomator2-driver\node_modules\follow-redirects\index.js:14:24)
npm ERR!     at ClientRequest.emit (node:events:514:28)
npm ERR!     at TLSSocket.socketErrorListener (node:_http_client:495:9)
npm ERR!     at TLSSocket.emit (node:events:514:28)
npm ERR!     at emitErrorNT (node:internal/streams/destroy:151:8)
npm ERR!     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
npm ERR!     at processTicksAndRejections (node:internal/process/task_queues:82:21)
npm ERR! [10:48:57] Downloading Chromedriver can be skipped by setting the'APPIUM_SKIP_CHROMEDRIVER_INSTALL' environment variable.

npm ERR! A complete log of this run can be found in: C:\Users\ty\AppData\Local\npm-cache\_logs\2023-08-18T02_48_27_627Z-debug-0.log`

This error always occurs when I install "appium driver install uiautomator2".

I don't understand. Why?

=================================
appium2 is not as convenient as appium-desktop.

appium-desktop download and run it directly.

appium2 also installs nodejs and appium in npm to install the driver.then download appium-inspector. Have some trouble

**Comment by KazuCocoa** [comment]
also, please update the description following by bug template. https://discuss.appium.io/ may also help to ask questions

**Comment by tianyi725** [comment]
> The chromedriver download got failed, maybe your network was not good to download content from Google? https://github.com/appium/appium-chromedriver
> 
> As the error message, `the'APPIUM_SKIP_CHROMEDRIVER_INSTALL` can help to skip the download in the installation time.

Thank you for your reply

I found in https://github.com/appium/appium-chromedriver installation method: NPM install appium - chromedriver
No, I set chromedriver_cdnurl.

Be downloading
npm install appium-chromedriver --chromedriver_cdnurl=http://npm.taobao.org/mirrors/chromedriver
Still failing.

Feel chromedriver_cdnurl=http://npm.taobao.org/mirrors/chromedriver this url is wrong.
Is there any other way to install it?


Also, this variable you're talking about, APPIUM_SKIP_CHROMEDRIVER_INSTALL
Where do I set it?


thank you
